torch/ao/quantization/fx/prepare.py

```diff
@@ -1129,38 +1200,136 @@ def insert_observers_for_model(
     #
     cache_for_no_tensor_check: Dict[Node, bool] = {}
 
-    inputs_seen_counter = 0
-    outputs_seen_counter = 0
-
     # first, populate the dtype map based only on qconfig and qhandler
     # this assumes:
     # graph inputs are fp32 by default, and int8 where overriden
     # other nodes output dtype is specified by the qconfig
     named_modules = dict(model.named_modules(remove_duplicate=False))
+
+    input_quantized_idxs: List[int] = prepare_custom_config.input_quantized_indexes
+    output_quantized_idxs: List[int] = prepare_custom_config.output_quantized_indexes
+    processed_nodes: Set[Node] = set()
+    # initalize target_dtype_info
+    DEFAULT_QCONFIG_FOR_TARGET_DTYPE_INFO = (
+        QConfig(
+            activation=PlaceholderObserver.with_args(dtype=torch.float32),
+            weight=PlaceholderObserver.with_args(dtype=torch.float32)
+        )
+    )
+    for node in model.graph.nodes:
+        node.meta["target_dtype_info"] = {
```

Comment:
does this assume one input/one output?  Thoughts on what happens when there are multiple inputs / output is not a node?

Reply:
currently this means we'll use these observers/fqs for all the input and outputs that need to be quantized